lib/Parser/Parse.cpp

```diff
@@ -3534,6 +3734,11 @@ ParseNodePtr Parser::ParsePostfixOperators(
                         if (pToken->tk == tkID && pToken->pid == wellKnownPropertyPids.eval && count > 0) // Detect eval
                         {
                             this->MarkEvalCaller();
+
+                            this->PushPidRef(wellKnownPropertyPids._this);
```

Comment:
>this->PushPidRef(wellKnownPropertyPids._this); [](start = 28, length = 46)

Hide behind something like "Reserve this"

Reply:
👍